tracks/x86-64-assembly/exercises/two-fer/mentoring.md

```diff
@@ -0,0 +1,93 @@
+### Reasonable solutions
+
+Using a hand-rolled loop:
+
+```nasm
+    movzx edx, byte [rdi]
+    test dl, dl
+    je .L2
+.L1:
+    add rdi, 1
+    mov byte [rsi], dl
+    add rsi, 1
+    movzx edx, byte [rdi]
+    test dl, dl
+    jne .L1
+.L2:
+```
+
+Using the `rep movsb` instructions:
+
+```nasm
+    mov rdi, rsi
+    lea rsi, [rel symbol]
+    mov rcx, symbol_len
+    rep movsb
+```
+
+### Common errors
+
+- Forgetting to save registers with callee-save status (`rbx`, `rbp`, `r12-r15`).
+- Forgetting return. A function declaration must end with `ret`.
+
+### Common suggestions
+
+- Move the string copy code into its own function or macro to avoid code duplication.
```

Comment:
I love these suggestions. It might be wise to note write them as "copyable" content and more like:

- If a student has code duplication, help them find a way to de-duplicate it, for example by having multiple functions or show them that macros exist and how they work.

We probably usually don't want to "give a way the solution".